actors/builtin/miner/miner_state.go

```diff
@@ -957,6 +957,30 @@ func (st *State) checkPrecommitExpiry(store adt.Store, sectors abi.BitField) (de
 	return depositToBurn, nil
 }
 
+func MinerEligibleForElection(store adt.Store, mSt *State, thisEpochReward abi.TokenAmount, minerActorBalance abi.TokenAmount, currEpoch abi.ChainEpoch) (bool, error) {
+	// IP requirements are met.  This includes zero fee debt
+	if !mSt.MeetsInitialPledgeCondition(minerActorBalance) {
+		return false, nil
+	}
+
+	// No active consensus faults
+	mInfo, err := mSt.GetInfo(store)
```

Comment:
Consider taking the info as a parameter instead of the store. The caller likely has it already.